lib/cuda/cudaCommand.hpp

```diff
@@ -97,6 +97,9 @@ class cudaCommand : public gpuCommand {
     virtual cudaEvent_t execute_base(cudaPipelineState& pipestate,
                                      const std::vector<cudaEvent_t>& pre_events);
 
+    virtual bool should_execute(cudaPipelineState& pipestate,
```

Comment:
Are we still going to need this with the ring buffer?

Reply:
Good point, we should probably remove this whole mechanism after the dust settles!  *Possibly* including removing the `cudaPipelineState` class; the buffer metadata objects will ideally hold any information that has to flow down the pipeline.

I don't really want to scrap this just yet, because so far this change hasn't broken the old MVP setup!